src/auth/de_auth_token.erl

```diff
@@ -0,0 +1,104 @@
+-module(de_auth_token).
+-include_lib("deliverly/include/deliverly.hrl").
+-include_lib("deliverly/include/log.hrl").
+-define(NAMESPACE, <<"delivery:auth:token:">>).
+
+-export([request_token/1, verify/2, verify/3]).
+
+request_token(Options) ->
+  random:seed(erlang:now()),
+  request_token_opt(merge(Options)).
+
+verify(#de_client{app = App}, Data) ->
+  Token = proplists:get_value(<<"token">>, Data),
+  case get_scope(Token) of
+    undefined ->
+      false;
+    <<"*">> ->
+      true;
+    Apps->
+      lists:member(atom_to_list(App), string:tokens(binary_to_list(Apps), ","))
+  end.
+
+verify(Client, Data, _Args) ->
+  verify(Client, Data).
+
+merge(Options) ->
+  Default = #{
+    expires_in => ulitos_app:get_var(deliverly, token_timeout, 60),
+    once => ulitos_app:get_var(deliverly, token_once, false),
+    scope => <<"*">>
+  },
+  maps:map(fun to_term/2, merge(Default, Options)).
+
+merge(Default, Options) when is_map(Options) ->
+  maps:map(fun(K, V) -> maps:get(K, Options, V) end, Default);
+
+merge(Default, Options) when is_list(Options) ->
+  maps:map(
+    fun(K, V) ->
+      proplists:get_value(K, Options, proplists:get_value(atom_to_binary(K, utf8), Options, V))
```

Comment:
Line 40 is too long:       proplists:get_value(K, Options, proplists:get_value(atom_to_binary(K, utf8), Options, V)).